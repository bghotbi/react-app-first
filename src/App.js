import './App.css';
import { NavBar } from './Components/NavBar';
import { Products } from './Components/Products';
import './index.css'


export const App = ({name, message}) => {
  return (
    <>
      <NavBar/>
      <div className="App">
         <h1><span>{message}</span>  <span>{name}</span> </h1> 
         <Products/>
      </div>
    </>
  );
}
