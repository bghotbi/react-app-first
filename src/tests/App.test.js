import { render, screen } from '@testing-library/react';
import renderer from "react-test-renderer";
import { App } from '../App';

jest.mock("../Components/NavBar", () => ({
  NavBar: () => {
    return <mock-navbar data-testid="navbar"/>;
  },
}));
jest.mock("../Components/Products", () => ({
  Products: () => {
    return <mock-products data-testid="products"/>;
  },
}));

describe('App component', () => {

  test('to mach snapshot', () => {
    const tree = renderer.create(<App name='bob' message='welcome test user'/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('it Renders', () => {
    const message = 'hello world!';
    const name = 'app';
    render(<App name={name} message={message}/>);
  });

  test('Render App with message and name', () => {
    const message = 'hello world!';
    const name = 'app';
    render(<App name={name} message={message}/>);
    expect(screen.getByText('hello world!')).toBeInTheDocument();
    expect(screen.getByText('app')).toBeInTheDocument();
  });

});
