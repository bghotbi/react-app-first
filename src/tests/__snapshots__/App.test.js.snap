// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component to mach snapshot 1`] = `
Array [
  <mock-navbar
    data-testid="navbar"
  />,
  <div
    className="App"
  >
    <h1>
      <span>
        welcome test user
      </span>
        
      <span>
        bob
      </span>
       
    </h1>
    <mock-products
      data-testid="products"
    />
  </div>,
]
`;
